/**
 * Copyright 2019 Huawei Technologies Co., Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto2";

message Checkpoint {
    message Value {
        required string tag = 1;
        required TensorProto tensor = 2;
    }
    repeated Value value = 1;
}


message TensorProto {
    // The shape of the tensor.
    repeated int64 dims = 1;
    // The type of the tensor.
    required string tensor_type = 2;
    // The data of the tensor.
    required bytes tensor_content = 3;
}
